components/usage/pkg/apiv1/billing.go

```diff
@@ -20,24 +21,31 @@ import (
 	"gorm.io/gorm"
 )
 
-func NewBillingService(stripeClient *stripe.Client, billInstancesAfter time.Time, conn *gorm.DB) *BillingService {
+func NewBillingService(stripeClient StripeClient, billInstancesAfter time.Time, conn *gorm.DB, usageClient v1.UsageServiceClient) *BillingService {
 	return &BillingService{
 		stripeClient:       stripeClient,
 		billInstancesAfter: billInstancesAfter,
 		conn:               conn,
+		usageClient:        usageClient,
 	}
 }
 
+type StripeClient interface {
```

Comment:
The reason why this changed is so that the Stripe client can be "mocked" in `billing-test.go`. I couldn't find another way to do it. I understand this is probably not expected, so I'm happy to talk about how to mock `GetUpcomingInvoice` in a more isolated way.